addon/modifiers/did-intersect.js

```diff
@@ -38,15 +47,38 @@ export default class DidIntersectModifier extends Modifier {
       return;
     }
 
-    let { onEnter, onExit, options } = this.args.named;
+    let { onEnter, onExit, maxEnter, maxExit, options } = this.args.named;
 
     assert('onEnter or/and onExit is required', onEnter || onExit);
 
-    if (onEnter) {
-      this.observerManager.addEnterCallback(this.element, onEnter);
+    // We should technically accept 0, in case of programmatic changes
+    if (maxEnter || maxEnter === 0) {
+      assert('maxEnter must be an integer', Number.isInteger(maxEnter));
+      this._maxEnterIntersections = maxEnter;
     }
-    if (onExit) {
-      this.observerManager.addExitCallback(this.element, onExit);
+
+    // We should accept 0 here too.
+    if (maxExit || maxExit === 0) {
+      assert('maxExit must be an integer', Number.isInteger(maxExit));
+      this._maxExitIntersections = maxExit;
+    }
+
+    if (onEnter && maxEnter !== 0) {
+      this.observerManager.addEnterCallback(this.element, (...args) => {
+        if (!maxEnter || this._numOfEnters < maxEnter) {
+          onEnter(...args);
+        }
```

Comment:
I think we should consider removing the callbacks once the max is reached, for performance reasons. Thoughts @chriskrycho?

Reply:
I don't see an equivalent `remove` callback hook in `intersection-observer-admin`. I don't think we would want to `unobserve` here.
Created an issue: https://github.com/snewcomer/intersection-observer-admin/issues/25

Reply:
Ok, so it seems like performance is not really impacted because, `addEnterCallback` and `addExitCallback` does not add event listeners, but rather it is a subset of values for a key in a weakmap, the key being the observed element. 

We can explore further in the related issue, but I think this might not be as big of a blocker as initially thought.

Reply:
Well, the impact is if there is *only* an enter callback or an exit callback but not both (or if both have reached their max), we want to ensure we stop observing at that point. We can do that on our end, though, because we know whether we have both. If you can implement that basic logic at least, that should be sufficient.